src/CodeGen_LLVM.h

```diff
@@ -70,6 +71,27 @@ class CodeGen_LLVM : public IRVisitor {
     /** Initialize internal llvm state for the enabled targets. */
     static void initialize_llvm();
 
+    /** Setup state for ad-hoc codegen. Used in JIT outside of compiling an entire Halide module.
```

Comment:
"TODO: This probably shouldn't exist" is not confidence-inspiring. Could you clarify? E.g. on init_for_codegen something like "only used directly by the wasm backend for ad-hoc JIT testing"

Reply:
Guilty as charged. Let me do another pass to see if I can make this less bad (perhaps consolidating with the existing argv-wrapper code if possible).